src/Components/Canvas/Canvas.tsx

```diff
@@ -1,69 +1,78 @@
 import React from 'react';
 import "./Canvas.css";
-import P5 from 'p5';
-import p5 from 'p5';
+// import P5 from 'p5';
+// import p5 from 'p5';
 import 'p5';
 import '../../Types/Figures';
 import { AnimatedFigure, CircleFigure, SquareFigure, TriangleFigure } from '../../Types/ProcessingFigures';
 import { CanvasSettings, SelectedShape, SelectedAnimation } from '../../Types/Figures';
+import P5Wrapper from 'react-p5-wrapper';
+import 'react-p5-wrapper';
 
-export default function Canvas(/* { settings: CanvasSetting}*/) {
-    let myP5: P5;
-    let myRef: React.RefObject<HTMLDivElement> = React.createRef();
+interface ICanvasProps {
+    canvasSettings?: CanvasSettings
+}
+
+interface ICanvasState {
 
-    /*
-     * TODO (delete this and read from props)
-     * Mock data 
-     */
-    let settings: CanvasSettings = {
-        selectedFigure: SelectedShape.Triangle,
-        selectedAnimation: SelectedAnimation.None,
-        reset: false
-    };
+}
 
-    let Sketch = (p : P5) => {
+export default class Canvas extends React.Component<ICanvasProps, ICanvasState> {
```

Comment:
I converted this component back to OO, because the myP5 = new p5(...) allocation can only be done a single time (otherwise multiple canvases will be drawn). I got around this issue by using the react-p5-wrapper library, but we might need the componentDidMount functions for similar canvas changes.